bundles/org.openhab.binding.opensmartcity/src/main/java/org/openhab/binding/opensmartcity/internal/OpenSmartCityWeatherHandler.java

```diff
@@ -102,9 +131,25 @@ private void updateSensorValues() {
                 updateStatus(ThingStatus.ONLINE);
 
                 String content = response.getContentAsString();
-                JsonObject jsonResponse = gson.fromJson(content, JsonObject.class);
-                logger.debug("Response: {}", jsonResponse.toString());
-                // JsonElement observations = jsonResponse.get("value").getAsJsonArray().get(0);
+                if (content != null) {
+
+                    JsonObject jsonResponse = gson.fromJson(content, JsonObject.class);
+                    if (jsonResponse != null) {
+                        logger.debug("Response: {}", jsonResponse.toString());
+                        JsonElement observations = jsonResponse.get("value").getAsJsonArray().get(0);
+                        String rs = observations.toString();
+                        JsonObject jsonResult = gson.fromJson(rs, JsonObject.class);
+                        if (jsonResult != null) {
+                            JsonElement result = jsonResult.get("Observations").getAsJsonArray().get(0);
+                            JsonObject valueObj = gson.fromJson(result.toString(), JsonObject.class);
+                            if (valueObj != null) {
+                                JsonElement value = valueObj.get("result");
+                                logger.debug("Value: {}", value);
```

Comment:
So can't you read the temperature and humidity here now? Why all these changes if we in the end still use fake values?

Reply:
Comes with next commit